src/olympia/blocklist/serializers.py

```diff
@@ -23,29 +23,32 @@ class Meta:
             'created',
             'modified',
             'addon_name',
+            'blocked',
+            'soft_blocked',
             'guid',
-            'min_version',
-            'max_version',
             'reason',
             'url',
-            'versions',
             'is_all_versions',
         )
 
-    def get_versions(self, obj):
+    def _get_blocked_for(self, obj, *, is_soft):
         if not hasattr(obj, '_blockversion_set_qs_values_list'):
             obj._blockversion_set_qs_values_list = sorted(
                 obj.blockversion_set.order_by('version__version').values_list(
```

Comment:
Could you not add soft to the query here? I'm not totally sure if that's valid in django but 

```python
obj.blockversion_set.filter(soft=is_soft)...
```

doesn't work? That would be better than filtering after the fact from a query perf perspective and much cleaner to look at.